lib/Controller/LoginController.php

```diff
@@ -419,6 +420,42 @@ private function provisionUser(string $userId, int $providerId, object $idTokenP
 		return $user;
 	}
 
+	/**
+	 * @NoCSRFRequired
+	 *
+	 * @param int $providerId
+	 * @param string $userId
+	 * @param string|null $displayName
+	 * @param string|null $email
+	 * @param string|null $quota
+	 * @return DataResponse
+	 */
+	public function apiCreateUser(int $providerId, string  $userId, ?string $displayName = null,
+								  ?string $email = null, ?string $quota = null): DataResponse {
+		$backendUser = $this->userMapper->getOrCreate($providerId, $userId);
+		$user = $this->userManager->get($backendUser->getUserId());
+
+		if ($displayName) {
+			if ($displayName !== $backendUser->getDisplayName()) {
+				$backendUser->setDisplayName($displayName);
+				$backendUser = $this->userMapper->update($backendUser);
+			}
+		}
+
+		if ($email) {
+			$user->setEMailAddress($email);
+		}
+
+		if ($quota) {
+			$user->setQuota($quota);
+		}
+
+		$this->userSession->setUser($user);
+		$this->userSession->completeLogin($user, ['loginName' => $user->getUID(), 'password' => '']);
```

Comment:
Thinking about it again, does this have any side effect when calling this API method through an existing user session? Maybe we should explicitly setup the FS only here: 

https://github.com/nextcloud/globalsiteselector/blob/1c72d573703406a3025c2d3b9727f8ac9d774b22/lib/UserBackend.php#L123-L129